GitUI/UserControls/RevisionGrid/Columns/GraphColumnProvider.cs

```diff
@@ -556,9 +552,32 @@ string GetLaneInfo(int x, int rowIndex)
 
                         if (node != null)
                         {
-                            if (laneInfoText.Length > 0)
+                            if (!node.Revision.IsArtificial)
                             {
-                                laneInfoText.AppendLine();
+                                laneInfoText.AppendLine(CommitIdColumnProvider.Format(node.Revision.Guid));
+
+                                var references = new References(node);
+
+                                if (references.CommittedTo.IsNotNullOrWhitespace())
+                                {
+                                    laneInfoText.AppendFormat("\nBranch: {0}", references.CommittedTo);
+                                    if (references.MergedWith.IsNotNullOrWhitespace())
+                                    {
+                                        laneInfoText.AppendFormat(" (merged with {0})", references.MergedWith);
+                                    }
+                                }
+
+                                laneInfoText.Append(Format("\nContained in branches: {0}\n", references.Branches));
+                                laneInfoText.Append(Format("\nContained in tags: {0}\n", references.Tags));
+
+                                string Format<T>(string format, IEnumerable<T> list)
```

Comment:
This doesn't belong here.  We have extension methods that can be used or modified.

Reply:
I would move it to there or even a list extensions class and call it ToCSV or something like it.